### PR TITLE
Add toggleable deconstruction

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
Clicking a thing marked for deconstruction will now stop it being deconstructed and interrupt drones who are on their way to do that.

### CODE REVIEW

File: code/datums/flock/flock.dm [hunk 1]
```diff
@@ -453,6 +453,12 @@
 		count++
 	return count
 
+/datum/flock/proc/toggleDeconstructionFlag(var/atom/target)
+	if(target in src.deconstruct_targets)
+		src.deconstruct_targets -= target
+	else
+		src.deconstruct_targets += target
+	src.updateAnnotations()
```
Comment (stonepillars) [inline code/datums/flock/flock.dm:461]: I'm certain pali or waka will complain that this isn't a macro

Comment (amylizzle) [reply code/datums/flock/flock.dm:461]: is there a macro for this?

Comment (stonepillars) [reply code/datums/flock/flock.dm:461]: No but proc calls are slow or something :shelterfrog:

Comment (FlameArrow57) [reply code/datums/flock/flock.dm:461]: togglePriorityTurfs() does this, plus see annotations changes I made for togglePriorityTurfs(), would be making similar changes once this is merged

**Pull request merged**